projects/bp-gallery/src/components/common/picture-gallery/PictureScrollGrid.tsx

```diff
@@ -77,10 +77,25 @@ const PictureScrollGrid = ({
   );
 
   const pictures: FlatPicture[] | undefined = useSimplifiedQueryResponseData(data)?.pictures;
+
+  const pictureIdToIndex = useMemo(() => {
+    if (!(queryParams instanceof Array)) {
+      return null;
+    }
+    const map = new Map<string, number>();
+    let index = 0;
+    for (const id of queryParams) {
+      map.set(id, index);
+      index++;
+    }
+    return map;
```

Comment:
Does this work here?
```suggestion
    const map = new Map(queryParams.map((id, index) => [id, index]))
```

Reply:
it could. I'll try